native/src/main/java/io/xlibb/pipe/thread/WorkerThreadPool.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.xlibb.pipe.thread;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Class to hold the static cachedThreadPool for the Pipe package.
+ */
+public class WorkerThreadPool {
+    private static final String THREAD_NAME = "bal-pipe-thread";
+
+    private WorkerThreadPool() {
+    }
+
+    public static final ExecutorService PIPE_EXECUTOR_SERVICE = new ThreadPoolExecutor(0, 500,
```

Comment:
This max pool number should be configurable, and 500 is a big number for the general case, IMO.

Reply:
IMO we could have used Java Cached threadpool executor [1] in this case 

[1] - https://www.baeldung.com/java-executors-cached-fixed-threadpool